NuRadioReco/modules/channelThermalNoiseAdder.py

```diff
@@ -0,0 +1,282 @@
+import astropy.coordinates
+import functools
+import json
+import numpy as np
+import os
+import scipy.constants
+import scipy.interpolate
+from scipy.interpolate import interp1d
+from urllib.request import urlretrieve
+import warnings
+
+import NuRadioReco.detector.antennapattern
+import NuRadioReco.framework.channel
+import NuRadioReco.framework.sim_station
+from NuRadioReco.modules.base.module import register_run
+from NuRadioReco.utilities import units, ice, geometryUtilities
+
+import logging
+logger = logging.getLogger('NuRadioReco.channelThermalNoiseAdder')
+
+
+class channelThermalNoiseAdder:
+    """
+    Module to generate thermal noise from ice
+    This class is a stripped version of channelGalacticNoiseAdder
+    Intead of using a skymap, the module uses pre generated effective antenna temperatures,
+    generated by Steffen's effective temperature code which was modified by Felix
+    (see NuRadioMC/examples/simulate_effective_ice_temperature )
+    """
+
+    def __init__(self):
+        self.__n_side = None
+        self.__antenna_pattern_provider = NuRadioReco.detector.antennapattern.AntennaPatternProvider()
+
+
+    @functools.lru_cache(maxsize=1024 * 32)
+    def get_cached_antenna_response(self, antenna_pattern, zen, azi, *ant_orient):
+        return antenna_pattern.get_antenna_response_vectorized(self.freqs, zen, azi, *ant_orient)
+
+
+    def get_temperature_at_depth(self, depth):
+        grip_dir = f"{os.path.dirname(os.path.abspath(__file__))}/../../NuRadioMC/data"
+        grip_file = "griptemp.txt"
+        grip_path = grip_dir + "/" + grip_file
+
+        if not os.path.exists(grip_path):
+            grip_url = "https://doi.pangaea.de/10.1594/PANGAEA.89007?format=textfile"
+            urlretrieve(grip_url, grip_path)
+
+        grip_temp = np.loadtxt(grip_path, skiprows=39)
+        depth_values = -grip_temp[:,0]
+        temperature_values = grip_temp[:,1]
+        profile = interp1d(depth_values,
+                           temperature_values,
+                           bounds_error=False,
+                           fill_value=(temperature_values[-1],temperature_values[0]))
+        return profile(depth)
```

Comment:
please use `NuRadioMC.utilities.attenuation.get_grip_temperature` instead. Make somewhere sure that this is only valid for greenland ice.

Reply:
maybe even with a warning if the detector site is not summit or so.